### ISSUE TITLE
Enhancement: Instead of strings, why not 'direct' conversions?

### ISSUE THREAD

**Issue by Crayder** [Gammix/Colors-Library-Include#1]
```
stock HexToInt(hexcolor, &r, &g, &b, &a)
{
    r = ( hexcolor >>> 24 ) & 0xFF;
    g = ( hexcolor >>> 16 ) & 0xFF;
    b = ( hexcolor >>> 8 ) & 0xFF;
    a = hexcolor & 0xFF;
}
stock IntToHex(r, g, b, a = 255) 
    return (r*16777216)+(g*65536)+(b*256)+a;
```

It's that simple. Your string conversion methods are exactly what I've tried avoiding and here is my answer.

Example: 

```
new r, g, b, a;
HexToInt(0x00FF80AA, r, g, b, a);
printf("%i, %i, %i, %i", r, g, b, a); // Result: "0, 255, 128, 170"

printf("%x", IntToHex(0, 255, 128, 170)); // Result: "FF80AA"
```

**Comment by Crayder** [comment]
```
ColorBetween(ColorA, ColorB, Float:Percent)
{
    new R1, G1, B1, A1, R2, G2, B2, A2;

    HexToInt(ColorA, R1, G1, B1, A1);
    HexToInt(ColorB, R2, G2, B2, A2);

    return IntToHex(
        floatround(Percent * (R2-R1)) + R1,
        floatround(Percent * (G2-G1)) + G1,
        floatround(Percent * (B2-B1)) + B1,
        floatround(Percent * (A2-A1)) + A1
    );
}
```

Example:

```
printf("%x", ColorBetween(0xFFFFFFFF, 0x00000000, 0.0)); // Result: "0"
printf("%x", ColorBetween(0xFFFFFFFF, 0x00000000, 1.0)); // Result: "FFFFFFFF"
```

**Comment by Gammix** [comment]
Looks good.

But what's wrong in string method?

EDIT: Actually my COLOR_Create is what you call IntToHex!

Updated the include.

**Comment by Crayder** [comment]
There's nothing wrong with the string method... But it's so much easier to use integers, there's now no need convert between strings just to get the hex color you want. :D

Also, if you want to be a bit more precise, you might want to add a clamp to ColorBetween. Clamp the percent to 0.0 - 1.0.

You could also add a integer detection (tagof trick), so 0-100 could the percent as well as 0.0-1.0.

**Comment by Gammix** [comment]
Yeah, and i am making my own colors list just have total ownership over the include.

**Comment by Crayder** [comment]
Why? The guy (original creator of the "colors.inc") didn't license them colors, and all he did was extract the colors from the X11 wiki page.

http://forum.sa-mp.com/showthread.php?t=152174

**Comment by Gammix** [comment]
Because i have added a wide colors range. Its was easy making them using a HTML color picker.

**Comment by Crayder** [comment]
Fair enough, but I think you should keep the X11 colors. Many people know their names already and they won't **need** to adjust to your's.

ALSO: [Badass List](https://docs.google.com/spreadsheets/d/1v3sF-hTZcCn78wG0gi-AijlU5gEMcVKIEoSshA8YhcI/edit?usp=sharing)
Source: https://en.wikipedia.org/wiki/List_of_colors:_A%E2%80%93F https://en.wikipedia.org/wiki/List_of_colors:_G%E2%80%93M https://en.wikipedia.org/wiki/List_of_colors:_N%E2%80%93Z
